src/dispatch/__tests__/DataPlaneClient.test.ts

```diff
@@ -132,4 +132,107 @@ describe('DataPlaneClient tests', () => {
             'd37eb756444ebf6f785233714d6d942f2b20f69292fb09533f6b69556eb0ff2b'
         );
     });
+
+    test('when the endpoint contains a path then the fetch reqest url contains the path prefix', async () => {
+        // Init
+        const endpoint = new URL(
+            `${Utils.AWS_RUM_ENDPOINT.toString()}${'prod'}`
+        );
+        const client: DataPlaneClient = new DataPlaneClient({
+            fetchRequestHandler: new FetchHttpHandler(),
+            beaconRequestHandler: new BeaconHttpHandler(),
+            endpoint,
+            region: Utils.AWS_RUM_REGION,
+            credentials: Utils.createAwsCredentials()
+        });
+
+        // Run
+        await client.sendFetch(Utils.PUT_RUM_EVENTS_REQUEST);
+
+        // Assert
+        const signedRequest: HttpRequest = (fetchHandler.mock
+            .calls[0] as any)[0];
+        expect(signedRequest.hostname).toEqual(Utils.AWS_RUM_ENDPOINT.hostname);
+        expect(signedRequest.path).toEqual(
+            `${endpoint.pathname}/appmonitors/application123/`
+        );
+    });
+
+    test('when the endpoint path contains a trailing slash then the fetch reqest url drops the trailing slash', async () => {
+        // Init
+        const endpoint = new URL(
+            `${Utils.AWS_RUM_ENDPOINT.toString()}${'prod/'}`
+        );
+        const client: DataPlaneClient = new DataPlaneClient({
+            fetchRequestHandler: new FetchHttpHandler(),
+            beaconRequestHandler: new BeaconHttpHandler(),
+            endpoint,
+            region: Utils.AWS_RUM_REGION,
+            credentials: Utils.createAwsCredentials()
+        });
+
+        // Run
+        await client.sendFetch(Utils.PUT_RUM_EVENTS_REQUEST);
+
+        // Assert
+        const signedRequest: HttpRequest = (fetchHandler.mock
+            .calls[0] as any)[0];
+        expect(signedRequest.hostname).toEqual(Utils.AWS_RUM_ENDPOINT.hostname);
+        expect(signedRequest.path).toEqual(
+            `${endpoint.pathname.replace(
```

Comment:
question: Seems that we will use this similar regex pattern to parse the part of the `pathname` in the main code - Could we put that into a helper method and place it in util, then (1) reuse in test cases (2) create a unit test for it?

Reply:
The test is verifying that the trailing slash is dropped. Re-using the same code would defeat the purpose of the test, as it would not detect a regression in that function.

Reply:
Good point